modules/library/main/src/main/java/org/geotools/filter/visitor/SimplifyingFilterVisitor.java

```diff
@@ -388,42 +389,68 @@ public Object visit(Id filter, Object extraData) {
         return validIdFilter;
     }
 
+    /**
+     * Return true iff there is a MatchAction for the negation of a multivalued binary comparison operation.
+     */
+    private boolean isNegatableMatchAction(MatchAction matchAction) {
+        return matchAction == MatchAction.ANY || matchAction == MatchAction.ALL;
```

Comment:
Are you completely sure filters cannot be built with a null matchaction? Maybe some by old code that predates the matchaction support?